File: gin-server/gin-blog-server/api/v1/blogset.go
package v1

import (
	"NyaLog/gin-blog-server/model"
	"NyaLog/gin-blog-server/service"
	"NyaLog/gin-blog-server/utils/errmsg"
	"net/http"

	"github.com/gin-gonic/gin"
)

// 更新博客设置
func UpdateBlogSet(c *gin.Context) {
	var data model.BlogSet
	_ = c.ShouldBindJSON(&data)
	err := service.UpdateBlogSet(&data)
	if err != errmsg.SUCCESS {
		c.JSON(http.StatusOK, gin.H{
			"code":    err,
			"message": errmsg.GetErrorMsg(err),
		})
	} else {
		c.JSON(http.StatusOK, gin.H{
			"code":    err,
			"message": "update success",
		})
	}
}

// 读取博客设置
func SeleBlogSet(c *gin.Context) {
	blogset, err := service.SeleBlogSet()
	if err != errmsg.SUCCESS {
		c.JSON(http.StatusOK, gin.H{
			"code":    err,
			"message": errmsg.GetErrorMsg(err),
		})
	} else {
		c.JSON(http.StatusOK, gin.H{
			"code":           err,
			"sitename":       blogset.Sitename,
			"sitecreatetime": blogset.Sitecreatetime,
			"sitebackground": blogset.Sitebackground,
			"aiswitch":       blogset.Aiswitch,
			"aicategory":     blogset.Aicategory,
			"commentswitch":  blogset.Commentswitch,
		})
	}
}

File: vue-client/src/stores/blogset.js
import { defineStore } from 'pinia'
import axios from 'axios'
import { reactive } from 'vue'

export const useBlogSetStore = defineStore('blogset', () => {
    const data = reactive({
        sitename: "一个神秘用户",
        sitecreatetime: "",
        sitebackground: ""
    })

    async function GetBlogInfo() {
        const response = await axios.get('/queryblogset');
        if (response.data.code === 200) {
            data.sitename = response.data.blogsetinfo.sitename;
            data.sitecreatetime = response.data.blogsetinfo.sitecreatetime
            data.sitebackground = response.data.blogsetinfo.sitebackground
        } else {
            console.log(response.data.message);
        }
    }
    return {GetBlogInfo, data}
})